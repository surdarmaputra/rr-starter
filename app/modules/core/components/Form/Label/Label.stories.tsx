import type { Meta, StoryObj } from '@storybook/react';

import { Label, type LabelProps } from './Label';

type Story = StoryObj<typeof Label>;

const meta: Meta = {
  component: Label,
  title: 'Core/Components/Form/Label',
};

export default meta;

export const Default: Story = {
  render: (args: LabelProps) => <Label {...args}>Sample label</Label>,
};
