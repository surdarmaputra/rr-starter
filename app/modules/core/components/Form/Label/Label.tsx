import * as LabelPrimitive from '@radix-ui/react-label';
import { cva, type VariantProps } from 'class-variance-authority';
import * as React from 'react';

import { cn } from '~/modules/core/libs/shadcn/utils';

const labelVariants = cva(
  'text-sm leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-letter-body',
);

export interface LabelProps
  extends React.ComponentPropsWithoutRef<typeof LabelPrimitive.Root>,
    VariantProps<typeof labelVariants> {
  secondaryText?: React.ReactNode;
}

const Label = React.forwardRef<
  React.ComponentRef<typeof LabelPrimitive.Root>,
  LabelProps
>(({ children, className, secondaryText, ...props }, ref) => (
  <LabelPrimitive.Root
    className={cn(labelVariants(), className)}
    ref={ref}
    {...props}
  >
    {children}
    {Boolean(secondaryText) && (
      <span className="ml-1.5 font-normal text-letter-caption">
        {secondaryText}
      </span>
    )}
  </LabelPrimitive.Root>
));
Label.displayName = LabelPrimitive.Root.displayName;

export { Label };
