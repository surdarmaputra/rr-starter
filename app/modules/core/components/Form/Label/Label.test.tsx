import { render, screen } from '@testing-library/react';
import { describe, expect, test } from 'vitest';

import { Label, type LabelProps } from './Label';

describe('Label', () => {
  function renderComponent(props?: LabelProps) {
    return render(
      <Label data-testid="Label-testid" {...props}>
        Sample label
      </Label>,
    );
  }

  test('renders with data-testid', async () => {
    renderComponent();
    await screen.findByTestId('Label-testid');
    await screen.findByText(/Sample label/);
  });

  test('renders with isOptional', async () => {
    renderComponent({ secondaryText: 'Optional ' });
    await screen.findByText(/Sample label/);
    await screen.findByText(/Optional/);
  });

  test('renders snapshot', async () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });
});
