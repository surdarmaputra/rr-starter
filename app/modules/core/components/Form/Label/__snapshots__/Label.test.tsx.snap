// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Label > renders snapshot 1`] = `
<div>
  <label
    class="text-sm leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-letter-body"
    data-testid="Label-testid"
  >
    Sample label
  </label>
</div>
`;
